platform/reworkd_platform/schemas/workflow/blocks/summarization_webhook.py

```diff
@@ -35,22 +46,51 @@ async def run(self) -> BlockIOBase:
         return SummaryWebhookOutput(**self.input.dict(), result=response)
 
 
-async def summarize_and_extract(prompt: str) -> str:
+def fetch_file(filename: str) -> BytesIO:
+    session = boto3.Session(profile_name="dev")
+    REGION = "us-east-1"
+    bucket_name = "test-pdf-123"
+    s3_client = session.client("s3", region_name=REGION)
+    response = s3_client.get_object(Bucket=bucket_name, Key=filename)
+    response_body = response["Body"].read()
+    bytesIO_file = BytesIO(response_body)
+
+    return bytesIO_file
+
+
+def convert_pdf_to_string(bytesIO_file: BytesIO) -> str:
+    pdf_reader = pypdf.PdfReader(bytesIO_file)
+    extracted_text = ""
+
+    # Iterate over each page in the PDF
+    for page_num in range(len(pdf_reader.pages)):
+        # Get the current page
```

Comment:
You can do something like:
```suggestion
    for page in pdf_reader.pages:
        ...
```